db-migrations/src/main/resources/db/migration/V2__arena-data.sql

```diff
@@ -1,27 +1,21 @@
-CREATE TYPE arena_operation_type AS ENUM (
-    'INSERT',
-    'UPDATE',
-    'DELETE'
-    );
-
-CREATE TYPE arena_ingest_status AS ENUM (
-    'NEW',
-    'INGESTED',
-    'RETRY',
-    'FAILED'
-    );
-
 CREATE TABLE arena_data
 (
-    id                  SERIAL PRIMARY KEY   NOT NULL,
-    table_name          VARCHAR              NOT NULL,
-    operation_type      arena_operation_type NOT NULL,
-    operation_pos       BIGINT               NOT NULL,
-    operation_timestamp TIMESTAMP            NOT NULL,
-    ingest_status       arena_ingest_status  NOT NULL DEFAULT 'NEW',
+    id                  SERIAL PRIMARY KEY NOT NULL,
+    table_name          VARCHAR            NOT NULL,
+    operation_type      VARCHAR            NOT NULL CHECK ( operation_type IN ('INSERT', 'UPDATE', 'DELETE')),
+    operation_pos       BIGINT             NOT NULL,
+    operation_timestamp TIMESTAMP          NOT NULL,
+    ingest_status       VARCHAR            NOT NULL DEFAULT 'NEW' CHECK ( ingest_status IN ('NEW', 'INGESTED', 'RETRY', 'FAILED', 'IGNORED')),
     ingested_timestamp  TIMESTAMP,
-    ingest_attempts     INT                  NOT NULL DEFAULT 0,
+    ingest_attempts     INT                NOT NULL DEFAULT 0,
     last_retry          TIMESTAMP,
     before              JSONB,
     after               JSONB
-)
+);
+
+CREATE UNIQUE INDEX arena_data_table_operation_type_operation_pos_idx on arena_data (table_name, operation_type, operation_pos);
```

Comment:
Hva brukes indexen til? operation_pos skal være unik, så man får ikke så mye ut av å legge til table_name, operation_type på indexen.

Reply:
Hvis det er for performance så er det ikke sikkert at det trengs siden det ikke blir gjort så mange queries mot tabellen

Reply:
Den trengs for on conflict i ArenaDataRepository, det må skje på en unique constraint. Visste ikke hva som skulle være unique så jeg tok den mest restriktive tanken og tok alle tre feltene

Reply:
Kan hende at operation_pos holder til det formålet

Reply:
Ja, det kan være, men vanskelig å vite siden jeg ikke har sett hvordan den wrapperen fungerer. Men om det er som du sier at operation_pos er unikt identifiserbart over tablename så kan jeg godt endre det